.github/workflows/pymeasure_CI.yml

```diff
@@ -10,16 +10,32 @@ jobs:
   docs_lint:
     name: Docs and Linting
     runs-on: "ubuntu-latest"
+    defaults:
+      run:
+        shell: bash -l {0}
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - name: Cache conda
+        uses: actions/cache@v2
+        env:
+          CACHE_NUMBER: 0  # Increase to reset the cache
+        with:
+          path: ~/conda_pkgs_dir
+          key: py3.8-ubuntu-latest-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('requirements/*.yml') }}
+      - name: Install pymeasure requirements
+        uses: conda-incubator/setup-miniconda@v2
         with:
           python-version: "3.8"
-      - name: Install docs dependencies
+          environment-file: requirements/pymeasure.yml
+          activate-environment: pymeasure
+          auto-update-conda: true
+          use-only-tar-bz2: true  # for caching to work properly
+      - name: Install dev requirements
```

Comment:
I was trying to avoid the duplication of requirements in multiple places. That allows PyMeasure users to only install the main environment to get the basic packages necessary. However, I agree that this requires additional time during install for the CI pieces. I'm open to combining them and just giving PyMeasure users everything if they want to use that environment. What do you think @bilderbuchi?

Good catch also on the pip installs over the conda ones. We should fix that either way.

Reply:
I guess that people who install from git master (and thus using/needing the requirements.yml) will probably don't mind a couple of packages more, or will even be more inclined to run the tests because they are troubleshooting something on the bleeding edge?

My hope is that if we give all in one go/one env file, the resolution also happens in one go and it does not install pip packages over conda ones, but we'll have to try (locally?) and check the logs/output.

Reply:
+1 on reducing duplication

Reply:
Sounds good. I merged them into 1 environment file and moved RTD to use that file as well.